internal/client/client.go

```diff
@@ -545,7 +545,9 @@ func (a *APIClient) SwitchContext(name string) error {
 	if err := a.config.SwitchContext(name); err != nil {
 		return err
 	}
-	if err := a.invalidateCache(); err != nil {
+	if !a.getConnOK() {
```

Comment:
I think we need to call checkConnectivity instead. if we are cool then we can invalidate the cache. 
Does this make sense?